src/main/java/randoop/generation/CoverageTracker.java

```diff
@@ -0,0 +1,200 @@
+package randoop.generation;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.jacoco.core.analysis.*;
+import org.jacoco.core.data.ExecutionDataStore;
+import org.jacoco.core.data.SessionInfoStore;
+import org.jacoco.core.instr.Instrumenter;
+import org.jacoco.core.runtime.IRuntime;
+import org.jacoco.core.runtime.LoggerRuntime;
+import org.jacoco.core.runtime.RuntimeData;
+
+/**
+ * Tracks the coverage of each method under test. Largely based on
+ * http://www.jacoco.org/jacoco/trunk/doc/examples/java/CoreTutorial.java
+ */
+public class CoverageTracker {
+  public static CoverageTracker instance = new CoverageTracker();
+
+  private final IRuntime runtime = new LoggerRuntime();
+  private final MemoryClassLoader memoryClassLoader;
+
+  private final ExecutionDataStore executionData = new ExecutionDataStore();
+  private final SessionInfoStore sessionInfos = new SessionInfoStore();
+
+  private final Instrumenter instrumenter;
+  private final RuntimeData data;
+
+  /** Map from method name to coverage details. */
+  private final Map<String, CoverageDetails> coverageDetailsMap = new HashMap<>();
+
+  /** Set of names of all the classes under test */
+  private Set<String> classesUnderTest = new HashSet<>();
+
+  /**
+   * Coverage details related to a single method under test. Tracks total number of branches and
+   * number of uncovered branches.
+   */
+  public static class CoverageDetails {
```

Comment:
I suggest calling this `BranchCoverage` instead of `CoverageDetails`, because it is more descriptive.